test/src/org/apache/jmeter/protocol/http/sampler/TestJmeterVariableSameUser.java

```diff
@@ -20,79 +20,76 @@
+        jmctx.setVariables(jmvars);        
         HTTPSamplerBase sampler = (HTTPSamplerBase) new HttpTestSampleGui().createTestElement();
-        jmvars.putObject(JMeterThread.IS_SAME_USER, false);
         CookieManager cookieManager = new CookieManager();
         cookieManager.setControlledByThread(true);
         sampler.setCookieManager(cookieManager);
-        HTTPHC4Impl hc = new HTTPHC4Impl(sampler);
-        hc.sample(new URL(url), "GET", true, 0);
+        sampler.setThreadContext(jmctx);
+        boolean res=(boolean) cookieManager.getThreadContext().getVariables().getObject("__jmv_SAME_USER");
         assertTrue("When test different user on the different iternation, the cookie should be cleared",
-                hc.getCookieManager().getClearEachIteration());
+                res);
     }
 
     @Test
-    public void testCookieManagerForSameUserOnIternation() throws MalformedURLException {
+    public void testCookieManagerForSameUserOnIternation() {
+        jmvars.putObject("__jmv_SAME_USER", false);
+        jmctx.setVariables(jmvars);        
         HTTPSamplerBase sampler = (HTTPSamplerBase) new HttpTestSampleGui().createTestElement();
         CookieManager cookieManager = new CookieManager();
         cookieManager.setControlledByThread(true);
         sampler.setCookieManager(cookieManager);
-        HTTPHC4Impl hc = new HTTPHC4Impl(sampler);
-        jmvars.putObject(JMeterThread.IS_SAME_USER, true);
-        hc.sample(new URL(url), "GET", true, 0);
+        sampler.setThreadContext(jmctx);
+        boolean res=(boolean) cookieManager.getThreadContext().getVariables().getObject("__jmv_SAME_USER");
         assertFalse("When test different user on the same iternation, the cookie shouldn't be cleared",
-                hc.getCookieManager().getClearEachIteration());
+                res);
     }
 
     @Test
-    public void testCacheManagerForDifferentUserOnSameIternation() throws MalformedURLException {
+    public void testCacheManagerForDifferentUserOnSameIternation() {
+        jmvars.putObject("__jmv_SAME_USER", false);
+        jmctx.setVariables(jmvars);        
         HTTPSamplerBase sampler = (HTTPSamplerBase) new HttpTestSampleGui().createTestElement();
         CacheManager cacheManager = new CacheManager();
         cacheManager.setControlledByThread(true);
         sampler.setCacheManager(cacheManager);
-        HTTPHC4Impl hc = new HTTPHC4Impl(sampler);
-        jmvars.putObject(JMeterThread.IS_SAME_USER, false);
-        hc.sample(new URL(url), "GET", true, 0);
-        assertTrue("When test different user on the different iternation, the cache should be cleared",
-                hc.getCacheManager().getClearEachIteration());
+        sampler.setThreadContext(jmctx);
+        boolean res=(boolean) cacheManager.getThreadContext().getVariables().getObject("__jmv_SAME_USER");
+        assertFalse("When test different user on the different iternation, the cache should be cleared",
```

Comment:
Test is not correct, you just test variable is set and not that cache is cleared